Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -75,6 +75,87 @@ public ITransformedNetwork getTransformedNetworkOfProject(String projectId)
         if (networkList == null) {
             return null;
         }
+        
+        ITransformedNetwork transformedNetwork = getTransformedNetworkusingNetworkList(networkList);
+        
+        return transformedNetwork;
+    }
+
+
+    /**
+	 * This method returns the transformation network in provided project for given conceptId.
+	 * @param projectIds
+	 * @param conceptId
+	 * @return ITransformedNetwork
+	 * @throws QuadrigaStorageException
+	 * @author suraj nilapwar
+	 */
+    @Override
+    public ITransformedNetwork getSearchTransformedNetwork(String projectId, String conceptId)
+        throws QuadrigaStorageException {
+        // get the transformed network and search for the concept id
+        List<INetwork> networkList = getNetworkList(projectId);
+
+        if (networkList == null) {
+            return null;
+        }
+
+        // get the transformed network of all the networks in a project
+        ITransformedNetwork transformedNetwork = getTransformedNetworkusingNetworkList(networkList);
+        // create finalnetwork using conceptId 
+        ITransformedNetwork finalTransformedNetwork = getFinalTransformedNetwork(transformedNetwork,conceptId);
+
+        return finalTransformedNetwork;
+    }
+
+    private List<INetwork> getNetworkList(String projectId) throws QuadrigaStorageException {
+        List<INetwork> networkList;
+        try {
+            networkList = networkManager.getNetworksInProject(projectId);
+        } catch (QuadrigaStorageException e) {
+            throw new QuadrigaStorageException("Database error while getting networks of a project" +
+                    " with id: " + projectId, e);
+        }
+
+        return networkList;
+    }
+    
+    /**
+	 * This method returns the transformation network in provided list of projects for given conceptId.
+	 * @param projectIds
+	 * @param conceptId
+	 * @return ITransformedNetwork
+	 * @throws QuadrigaStorageException
+	 * @author suraj nilapwar
```

Comment:
Still doesn't make sense. What is a "transformation network in provided project"? 1. there is not just one provided project, but several project ids. 2. do you mean "for the provided project"? 3. What is a transformation network for a given concept id?

You need to describe what your method is doing. It is searching the networks of the projects specified by the project ids, right? So you should say that. And what it is doing once it searched it.